src/metabase/api/app.clj

```diff
@@ -108,7 +153,25 @@
                         (cond-> {:page_id ["scaffold-target-id" "page" table-id page-type]}
                           (= page-type "detail") (assoc :indent 1 :hidden true)))}
      :cards (for [table-id table-ids
-                  :let [table (get table-id->table table-id)]
+                  :let [table (get table-id->table table-id)
+                        order-by-field-id (->> table
+                                               :fields
+                                               (keep (fn [field]
+                                                       (let [field-name (str/lower-case (:name field))]
+                                                         (cond
+                                                           (= field-name "updated_at")
+                                                           [0 (:id field)]
+
+                                                           (= field-name "created_at")
+                                                           [1 (:id field)]
+
+                                                           (and (= :type/PK (:semantic_type field))
+                                                             (isa? (:base_type field) :type/Number))
+                                                           [2 (:id field)]))))
+
+                                               sort
+                                               first
```

Comment:
nit: we only need the least element, but we sort the whole list.

Reply:
I'm not sure I follow, we don't know the order of fields, so we could get `[[2 123] [0 432] [1 688]]` The tuples coming out of the `cond` are `[priority field-id]`. We want to use the updated_at field (0), so the sort uses the priority and then pulls out the field-id. I could rewrite this with `sort-by` and `{:priority 0 :field-id 432}` if it would be more clear?

Reply:
I just mean that the function in `keep` determines the ordering. A `max-key` with a similar function could pick the element we are looking for.

But this is a nit because the input is probably not large and result won't be more readable other than it does more directly what we need. (This is because `max-key` doesn't work without any arguments and you would have to produce a number as key instead of a tuple. I'm afraid medley doesn't help with this either.)